src/IO/WriteHelpers.h

```diff
@@ -317,6 +317,8 @@ void writeAnyEscapedString(const char * begin, const char * end, WriteBuffer & b
                     writeChar('t', buf);
                     break;
                 case '\0':
+                    if constexpr (stop_at_first_nul)
```

Comment:
I recommend not doing this.

It's much better to put zero characters in a string explicitly than doing a silent data loss.

Reply:
This is a bit delicate and tricky... For SQLite - no problem, we can insert `abc␀123` (using `␀` for the c-style `\0` as otherwise things get confusing):
```sql
-- in sqlite cli:
-- Not the prettiest - but works as expected (to an extent: https://www.sqlite.org/nulinstr.html)
INSERT INTO t0 VALUES ('abc'||char(0)||'123');
-- This does not work as expected - we end up with 'abc backslash 0 123'
INSERT INTO t0 VALUES ('abc\0123');
```

For PostgreSQL `␀` withing a string is not allowed:
```sql
-- in psql console
INSERT INTO t0 VALUES ('abc'||chr(0)||'123');
ERROR: null character not permitted

-- And this one does not work as expected - we end up with 'abc backslash 0 123'
INSERT INTO t0 VALUES ('abc\0123');
```

So... the existing current solution does not work - as it corrupts the string with `backslash zero` in place of `␀`.

I've added another commit that shows a bit more what could be done:
- allow `␀` for SQLite (with the clunky escaping as shown above)
- truncate for PostgreSQL